File: src/main/java/com/violeth/blockjs/blockjs/binds/SelectionBind.java
package com.violeth.blockjs.blockjs.binds;

import org.bukkit.util.BlockVector;
import java.io.File;
import java.util.ArrayList;
import java.util.List;

/** Optional WorldEdit block selection support
 * TODO: It was just a prototype */
public class SelectionBind {
    public BlockVector[] selection;

    public List<File> scripts;

    public SelectionBind(BlockVector[] selection, File script) {
        this.selection = selection;

        this.scripts = new ArrayList<>();

        this.scripts.add(script);
    }
}

File: plugin/src/main/kotlin/io/github/blockjs/plugin/binds/SelectionBind.kt
package io.github.blockjs.plugin.binds

import org.bukkit.util.BlockVector
import java.io.File

/** Optional WorldEdit block selection support
 * TODO: It was just a prototype  */
class SelectionBind(selection: Array<BlockVector>, script: File) {
    var selection: Array<BlockVector>
    var scripts: MutableList<File>

    init {
        this.selection = selection
        scripts = mutableListOf()
        scripts.add(script)
    }
}
